geopandas/io/sql.py

```diff
@@ -43,9 +46,20 @@ def read_postgis(sql, con, geom_col='geom', crs=None, index_col=None,
     if geom_col not in df:
         raise ValueError("Query missing geometry column '{}'".format(geom_col))
 
+    def load_geom(x):
+        return shapely.wkb.loads(str(x), hex=hex_encoded)
     wkb_geoms = df[geom_col]
+    df[geom_col] = GeoSeries(wkb_geoms.apply(load_geom))
 
-    s = wkb_geoms.apply(lambda x: shapely.wkb.loads(binascii.unhexlify(x.encode())))
-    df[geom_col] = GeoSeries(s)
+    if crs is None:
+        def get_srid(x):
+            return shapely.geos.lgeos.GEOSGetSRID(x._geom)
+        unique_srids = df[geom_col].apply(get_srid).unique()
+        # only set a srs for frame if all polygons have the same one
+        if len(unique_srids) == 1:
+            unique_srid = unique_srids[0]
+            # if no defined SRID in geodatabase, returns SRID of 0
+            if unique_srid != 0:
+                crs = {"init": "epsg:{}".format(unique_srid)}
```

Comment:
The addition of `assert df.crs is None` in `test_read_postgis_default` was for this line.